demo/search/search.js

```diff
@@ -1,3 +1,104 @@
+var typeIcons = {
+   amenity: {
+     arts_centre: 'arg-gallery',
+     atm: 'marker',
+     bank: 'bank',
+     bar: 'bar',
+     bicycle_parking: 'bicycle_parking',
+     bicycle_rental: 'bicycle_rental',
+     bus_station: 'bus',
+     cafe: 'cafe',
+     car_wash: 'marker',
+     cinema: 'cinema',
+     clinic: 'marker',
+     courthouse: 'marker',
+     compressed_air: 'compressed_air',
+     dentist: 'dentist',
+     doctors: 'doctor',
+     fast_food: 'fast-food',
+     ferry_terminal: 'ferry',
+     fire_station: 'fire-station',
+     fuel: 'fuel',
+     hospital: 'hospital',
+     kindergarten: 'scooter',
+     library: 'library',
+     pharmacy: 'pharmacy',
+     place_of_worship: 'place_of_worship',
+     police: 'police',
+     post_office: 'post',
+     pub: 'beer',
+     restaurant: 'restaurant',
+     school: 'school',
+     shelter: 'shelter',
+     theatre: 'theatre',
+     townhall: 'town-hall',
+     college: 'college',
+     university: 'college',
+     default: 'marker'
+   },
+   man_made: {
+     tower: 'marker',
+     default: 'marker'
+   },
+   tourism: {
+     attraction: 'attraction',
+     information: 'information',
+     camp_site: 'campsite',
+     caravan_site: 'campsite',
+     chalet: 'home',
+     hostel: 'home',
+     motel: 'home',
+     guest_house: 'home',
+     hotel: 'lodging',
+     museum: 'museum',
+     picnic_site: 'picnic-site',
+     viewpoint: 'viewpoint',
+     zoo: 'zoo',
+     theme_park: 'theme_park',
+     default: 'marker'
+   },
+   historic: {
+     archaeological_site: 'hillfort',
+     memorial: 'memorial',
+     manor: 'marker',
+     monastery: 'marker',
+     default: 'marker'
+   },
+   railway: {
+     station: 'rail',
+     default: 'marker'
+   },
+   aeroway: {
+     terminal: 'airport',
+     helipad: 'heliport',
+     aerodrome: 'airfield',
+     default: 'marker'
+   },
+   shop: {
+     alcohol: 'alcohol-shop',
+     car_repair: 'marker',
+     bakery: 'bakery',
+     bicycle: 'bicycle',
+     clothes: 'clothing-store',
+     supermarket: 'grocery',
+     mall: 'grocery',
+     department_store: 'grocery',
+     convenience: 'shop',
```

Comment:
Naikinti.

Reply:
Kodėl naikinti?

Reply:
Todėl kad kaip ir minėjau icona sutampa su default.
`shop.convenience == shop.default`

Reply:
Na šioje vietoj nesutikčiau, tegul būna, kad matytųsi sąmoningas tokių piktogramų priskyrimas konkretiems tipams.

Reply:
Tai pagal tavo logika reiktu aprasyti visus osm value galimus variantus https://wiki.openstreetmap.org/wiki/Key:shop, kas mano manymu būtu perteklinė informacija.

Reply:
Ne, aprašome tik tuos, apie kuriuos pagalvojome duomenų užklausose kurdami vektorines kaladėles. Kad būtų daugmaž 1:1 tarp logikos sql'e ir js.

Reply:
Nu jo, bet matai tavo uzklausos tegolai, ir mūsu paieška nieko bendro neturi.
Jei kažko tu neištrauki iš mūsu DBkurdamas layer, nesiskaito kad nebus to ir paieškoje, paieškoje bus daugiau duomenų, nei mes vaizduojame savo žemėlapyje.

Reply:
Kodo prasme neturi, o visos sistemos veikimo prasme - turi.
Kodas yra vedamasis, o ne vedantysis. Kodas yra tik apipavidalinimas, pagal jį sprendimai nedaromi.